### ISSUE TITLE
Refresh tokens flow with silent renew is stuck at isProbablyStuck v12.0.3

### ISSUE THREAD

**Issue by viswadv** [damienbod/angular-auth-oidc-client#1274]
Hello Damien,
After a sequence of few successful refresh tokens, suddenly a response from the refresh tokens request is never updated to the storage and control goes to the isProbablyStuck condition where elapsedTimeInMilliseconds value is around 59s.

There is no pattern for this, sometimes it happens for the first request and sometimes it happens randomly for some nth request. When this happens the refresh token request is again made with the same old used refresh token which our server rejects as it's only one-time use and this continues for 2/3 times and logout is triggered.

Please help me identify the issue and is it possible to add 
this.storagePersistanceService.write('authnResult', callbackContext.authResult); in Step 2 Refresh token switchmap?

A quick response would be appreciated.

**Comment by ppatil27** [comment]
@damienbod I am also having same issue where refresh token api with latency more than 2seconds is not being handled by library. It is not yet reaching to step 3 and got failed after retry. If api takes more than 2 seconds to response back, periodic check which is running every 4 seconds its getting into in isProbableStuck condition.

**Comment by viswadv** [comment]
No, it doesn't help. I think there is some issue in handling the delayed response of refresh tokens. I pointed the sample to my app server and randomly at some point a particular refresh token response is not treated the way it should and then a sequence of retries with the same old refresh token is made which is rejected by the server as its one-time use only. Please look into it, this is happening in our production.

**Comment by MarkCuypersPpw** [comment]
I can confirm I found the problem on my side. I updated this:

`  AuthModule.forRoot({
    storage: new LocalStorageAdapter(),
    config: {
      <...config here...>
   }
`
to this:

`  AuthModule.forRoot({
    config: {
      storage: new LocalStorageAdapter(),
      <...config here...>
    }
`
Then it started working again.
It seems the localstorage adapter wasn't seen by angular-auth-oidc-client, resulting in a fallback to the default session-storage. Since we are creating an app which opens detail-screens in a new browser tab, it was crucial to us that the security parameters are stored in something that is shared based on origin.
